src/presetsController.ts

```diff
@@ -955,17 +958,88 @@ export class PresetsController {
         return presetsFile;
     }
 
+    private populatePrivatePresetsFields(presetsFile: preset.PresetsFile | undefined, file: string) {
+        if (!presetsFile) {
+            return;
+        }
+
+        presetsFile.__path = file;
+        const setFile = (presets: preset.Preset[] | undefined) => {
+            if (presets) {
+                for (const preset of presets) {
+                    preset.__file = presetsFile;
+                }
+            }
+        };
+        setFile(presetsFile.configurePresets);
+        setFile(presetsFile.buildPresets);
+        setFile(presetsFile.testPresets);
+    }
+
+    private async mergeIncludeFiles(rootPresetsFile: preset.PresetsFile | undefined, presetsFile: preset.PresetsFile | undefined, file: string, mergedFiles: Set<string> = new Set()): Promise<void> {
+        if (!rootPresetsFile || !presetsFile || !presetsFile.include) {
+            return;
+        }
+
+        // Merge the includes in reverse order so that the final presets order is correct
+        for (let i = presetsFile.include.length - 1; i >= 0; i--) {
+            const fullIncludePath = path.normalize(path.resolve(path.dirname(file), presetsFile.include[i]));
+
+            // Do not include files more than once
+            if (mergedFiles.has(fullIncludePath)) {
+                continue;
+            }
+            mergedFiles.add(fullIncludePath);
+
+            const includeFileBuffer = await this.readPresetsFile(fullIncludePath);
+            if (!includeFileBuffer) {
+                log.error(localize('included.presets.file.not.found', 'Included presets file {0} cannot be found', fullIncludePath));
```

Comment:
I introduced this new localized string. This would need to be localized and added to the translation files.

Reply:
This will happen automatically after the PR is committed.